src/Augurk.CommandLine/Plumbing/AugurkHttpClientFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Augurk.CommandLine.Options;
+using System;
+using System.ComponentModel.Composition;
+using System.Net.Http;
+using System.Text;
+
+namespace Augurk.CommandLine.Plumbing
+{
+    internal static class AugurkHttpClientFactory
+    {
+        public static HttpClient CreateHttpClient(SharedOptions options)
+        {
+            if (options.UseIntegratedSecurity)
+            {
+                var username = System.Security.Principal.WindowsIdentity.GetCurrent().Name;
+                Console.WriteLine($"Using integrated security with user {username} to access the Augurk API's.");
+
+                var handler = new HttpClientHandler
+                {
+                    UseDefaultCredentials = true,
+                    PreAuthenticate = true,
+                };
+                var secureClient = new HttpClient(handler);
+                return secureClient;
+            }
+
+            var client = new HttpClient();
+
+            if (options.UseBasicAuthentication)
+            {
+                if (string.IsNullOrEmpty(options.BasicAuthnUsername))
+                {
+                    Console.Error.WriteLine("When using basic HTTP authentication, you must specify a username and (optionally) password)");
+                    System.Environment.Exit(-1);
+                }
+
+                var byteArray = Encoding.ASCII.GetBytes($"{options.BasicAuthnUsername}:{options.BasicAuthnPassword}");
```

Comment:
If the password is optional, shouldn't the presence of the : literal be dependent on the value of options.BasicAuthnPassword?